internal/wire/records_test.go

```diff
@@ -0,0 +1,72 @@
+package wire
+
+import (
+	"bufio"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func fetchRecords(recordsPath string) ([]testCase, error) {
+
+	// Fetch resursively every file path with ".bin" extension from recordsPath directory
+	var recordFiles []string
+	err := filepath.WalkDir(recordsPath, func(path string, entry fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if filepath.Ext(entry.Name()) == ".bin" {
+			recordFiles = append(recordFiles, path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Read every record file, parse their content to wire messages
+	// and store them in the testCase struct
+	var resMsgs []testCase
+	for _, path := range recordFiles {
+		f, err := os.OpenFile(path, os.O_RDONLY, 0o666)
```

Comment:
Just `os.Open`. And `f ` is never closed.

Reply:
I don't understand, I open reader for every file how `os.Open` would differ from that?

Reply:
os.Open has less parameters :)